File: Python/stage2/S2_FOR_04.py
# S2_FOR_04 — Count occurrences

def count_occurrences(values):

    counts = {}

    for v in values:

        if v in counts:

            counts[v] += 1

        else:

            counts[v] = 1

    return counts


# Tests

print(count_occurrences(["a", "b", "a", "c", "b", "a"]))
 

File: js/stage2/S2_FOR_04.js
// S2_FOR_04 — Count occurrences
function countOccurrences(values) {
  const result = {};
  for (const v of values) {
    result[v] = (result[v] || 0) + 1;
  }
  return result;
}
// Test
console.log(countOccurrences(["a", "b", "a", "c", "b", "a"]));
